noria/src/data.rs

```diff
@@ -243,21 +269,43 @@ impl Hash for DataType {
     }
 }
 
+impl From<i128> for DataType {
+    fn from(s: i128) -> Self {
+        if s < 0 {
+            DataType::BigInt(s as i64)
```

Comment:
Will these types of casts always succeed? There are negative values in the `i128` range that won't fit into an `i64`, right?

(Below cast from the positive `i128` to `u64` should be fine in that regard, though, as `u64` covers the entire positive range of `i128`.)

Reply:
I _think_ this is fine… others than a value coming from a stateful operator… But that would have been an issue before too, right?

Reply:
Added a check for `s >= i64::MIN` anyways…

Reply:
:+1: